compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1684,19 +1684,21 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         true
     }
 
+    // return true if hit type references_error
     fn point_at_returns_when_relevant(
         &self,
         err: &mut Diagnostic,
         obligation: &PredicateObligation<'tcx>,
-    ) {
+    ) -> bool {
```

Comment:
It's not obvious that this function returns a bool that is true if the type references error. 

could you change this `err: &mut DiagnosticBuilder<'tcx, ErrorGuaranteed>`, and then you can `delay_as_bug` inside the body?

Reply:
that's a great suggestion, and I changed to `DiagnosticBuilder `from Diagnostic for `invokcing `delay_as_bug inside `point_at_returns_when_relevant()`